src/multi-locale.ts

```diff
@@ -217,8 +250,20 @@ export function generateLocalizedAssets<LocalizedData>(
 					sourceString,
 					fileNameTemplatePlaceholder,
 				);
+				const contentHashLocations = [...contentHashMap.entries()]
+					.flatMap(([hash, hashesByLocale]) => findSubstringLocations(sourceString, hash)
```

Comment:
FYI `findSubstringLocations` was designed to find every match of a substring in one iteration when I saw there was a surprisingly large performance hit when doing many string searches (asset source can be very large).

I just benchmarked this against a RegExp approach where the pattern contains all the hashes (like RealContentHashPlugin) and it's still significantly performant 🤯: https://jsben.ch/q0eng

```ts
function findPatternLocations(string, pattern) {
  pattern.lastIndex = 0;
  let indices = [];
  let match;
  while ((match = pattern.exec(text))) {
    indices.push(match.index);
  }
  return indices;
}
```